docs/src/main/asciidoc/hibernate-orm-panache-kotlin.adoc

```diff
@@ -0,0 +1,204 @@
+
+[source,kotlin]
+----
+@Entity
+class Person : PanacheEntity {
+    companion object: PanacheCompanion<Person> {  // <1>
+        fun findByName(name: String) = find("name", name).firstResult()
+        fun findAlive() = list("status", Status.Alive)
+        fun deleteStefs() = delete("name", "Stef")
+    }
+
+    lateinit var name: String  // <2>
+    lateinit var birth: LocalDate
+    lateinit var status: Status
+}
+----
+<1> The companion object holds all the methods not related to a specific instance allowing for general management and
+querying bound to a specific type.
+<2> Here there are options, but we've chosen the `lateinit` approach.  This allows us to declare these fields as non-null
+knowing they will be properly assigned either by the constructor (not shown) or by hibernate loading data from the
+database.
+
+NOTE: Note that these types differ from the Java types mentioned in those tutorials.  For kotlin support, all the Panache
+types will be found in the `io.quarkus.hibernate.orm.panache.kotlin` package.  This subpackage allows for the distinction
+between the Java and Kotlin variants and allows for both to be used unambiguously in a single project.
+
+In the Kotlin version, we've simply moved the bulk of the link:https://www.martinfowler.com/eaaCatalog/activeRecord.html[`entity pattern`]
+functionality to the `companion object`.  Apart from this slight change, we can then work with our types in ways that map easily
+from the Java side of world.
+
+
+== Using the repository pattern
+
+
+=== Defining your entity
+
+When using the repository pattern, you can define your entities as regular JPA entities.
+[source,kotlin]
+----
+@Entity
+class Person {
+    @Id
+    @GeneratedValue
+    var id: Long? = null;
+    lateinit var name: String
+    lateinit var birth: LocalDate
+    lateinit var status: Status
+}
+----
+
+=== Defining your repository
+
+When using Repositories, you get the exact same convenient methods as with the active record pattern, injected in your Repository,
+by making them implement `PanacheRepository`:
+
+[source,kotlin]
+----
+class PersonRepository: PanacheRepository<Person> {
+     fun findByName(name: String) = find("name", name).firstResult()
+     fun findAlive() = list("status", Status.Alive)
+     fun deleteStefs() = delete("name", "Stef")
+}
+----
+
+All the operations that are defined on `PanacheEntityBase` are available on your repository, so using it
+is exactly the same as using the active record pattern, except you need to inject it:
+
+[source,kotlin]
+----
+@Inject
+lateinit var personRepository: PersonRepository
+
+@GET
+fun count() = personRepository.count()
+----
+
+=== Most useful operations
+
+Once you have written your repository, here are the most common operations you will be able to perform:
+
+[source,kotlin]
+----
+// creating a person
+var person: Person? = Person()
```

Comment:
Isn't it a bit weird to declare it as a nulltable type here? I guess we re-use the variable later, but still…

Reply:
Yeah, the only reason is the reuse below there.  But I'll tweak it.